src/OrbitGl/Track.cpp

```diff
@@ -161,11 +161,7 @@ void Track::Draw(Batcher& batcher, TextRenderer& text_renderer, uint64_t current
   }
 
   // Collapse toggle state management.
-  if (!this->IsCollapsible()) {
-    collapse_toggle_->SetState(TriangleToggle::State::kInactive);
-  } else if (collapse_toggle_->IsInactive()) {
-    collapse_toggle_->ResetToInitialState();
-  }
+  collapse_toggle_->SetIsCollapsible(this->IsCollapsible());
```

Comment:
Why we need this line?

Reply:
That is exactly the point when we tell the triangle to be collapsible or not. Unfortunately, we do not have an event system for tracks yet to get notified when the track itself becomes collapsible or not (which is determined differently for the subtracks, e.g. the timer track checks if there are timers). Currently, we basically pull that state every draw call. In the future we should probably use some event system (i.e. at the point when the state changes, notify the toggle).

But that is out of scope for this PR.